packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1360,3 +1361,103 @@ class KeepAlive extends ParentDataWidget<KeepAliveParentDataMixin> {
     properties.add(DiagnosticsProperty<bool>('keepAlive', keepAlive));
   }
 }
+
+// Return a Widget for the given Exception
+Widget _createErrorWidget(Object exception, StackTrace stackTrace) {
+  final FlutterErrorDetails details = FlutterErrorDetails(
+    exception: exception,
+    stack: stackTrace,
+    library: 'widgets library',
+    context: ErrorDescription('building'),
+  );
+  FlutterError.reportError(details);
+  return ErrorWidget.builder(details);
+}
+
+/// Set a cross axis extent constraint to the child Sliver. By default, slivers will try to
+/// fill up the available cross axis extent, but this parent data widget will override that
+/// behavior.
+class SliverConstrainedCrossAxis extends ParentDataWidget<SliverFlexParentData> {
+  /// Constructor.
+  const SliverConstrainedCrossAxis({
+    super.key,
+    required this.extent,
+    required Widget sliver,
+  }): super(child: sliver);
+  
+  /// Extent to constrain cross axis.
+  final double extent;
+
+  @override
+  void applyParentData(RenderObject renderObject) {
+    assert(renderObject.parentData is SliverFlexParentData);
+    final SliverFlexParentData parentData = renderObject.parentData! as SliverFlexParentData;
+    bool needsLayout = false;
+
+    if (parentData.extent != extent) {
+      parentData.extent = extent;
+      needsLayout = true;
+    }
+
+    if (needsLayout) {
+      final AbstractNode? targetParent = renderObject.parent;
+      if (targetParent is RenderObject) {
+        targetParent.markNeedsLayout();
+      }
+    }
+
+  }
+
+  @override
+  Type get debugTypicalAncestorWidgetClass => SliverCrossAxisGroup;
+}
+
+/// Set a flex value to determine how much remaining size to allocate to sliver.
```

Comment:
What size? And in which dimension? In which context would you use this? Is it specific to the cross axis sliver group only? Does it only flex the cross axis extent?